tartiflette/engine.py

```diff
@@ -284,17 +353,16 @@ def __init__(
         :rtype: Dict[str, Any]
         """
         document, errors = parse_and_validate_query(query, self._schema)
-        if errors:
-            return await self._build_response(errors=errors)
 
-        return await execute(
+        # Goes through potential schema directives and finish in self._perfom_query
+        return await self._query_executor(
             self._schema,
             document,
-            self._build_response,
-            initial_value,
+            errors,
+            operation_name,
             context,
             variables,
-            operation_name,
+            initial_value,
```

Comment:
You should filled in the `context_coercer` parameter in order to forward it to potentials arguments directives:

```suggestion
            initial_value,
            context_coercer=context,
```